### PR TITLE
Add self element to descendants lookup

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This PR adds self element to the list of possible descendants matches.

The problem has been discovered while creating tests for https://github.com/facebook/WebDriverAgent/pull/612

### CODE REVIEW

File: WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m [hunk 1]
```diff
@@ -73,29 +73,35 @@ - (XCElementSnapshot *)fb_lastSnapshot
     return @[];
   }
   NSArray<NSNumber *> *matchedUids = [snapshots valueForKey:FBStringify(XCUIElement, wdUID)];
+  NSMutableArray<XCUIElement *> *matchedElements = [NSMutableArray array];
+  if ([matchedUids containsObject:@(self.wdUID)]) {
+    if (1 == snapshots.count) {
+      return @[self];
```
Comment (marekcirkos) [inline WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: Why you assume self matches the request here?

Comment (mykola-mokhnach) [inline WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: _snapshots_ array defines how many elements should be returned. The situation where single snapshot matched to multiple elements is not possible. This means if we only need to match one snapshot and the uid of this snapshot matched to the uid of self element then the result should be single self element

Comment (marekcirkos) [inline WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: Can you add test case for this?

Comment (mykola-mokhnach) [inline WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: done

**Pull request closed without merging**